src/Sarif/Core/Run.cs

```diff
@@ -217,5 +217,50 @@ public bool ShouldSerializeColumnKind()
         public bool ShouldSerializeLogicalLocations() { return this.LogicalLocations.HasAtLeastOneNonDefaultValue(LogicalLocation.ValueComparer); }
 
         public bool ShouldSerializeNewlineSequences() { return this.NewlineSequences.HasAtLeastOneNonNullValue(); }
+
+        private static readonly Dictionary<string, FailureLevel> PoliciesCache = new Dictionary<string, FailureLevel>();
+
+        public void ComputePolicies(/*enum?*/)
```

Comment:
>public [](start = 8, length = 6)

make this an internal static for testing, make it accept an enumerable policies collection